plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/IcebergPositionDeletePageSink.java

```diff
@@ -0,0 +1,189 @@
+        this.writer = fileWriterFactory.createPositionDeleteWriter(new Path(outputPath), POSITION_DELETE_SCHEMA, jobConf, session, hdfsContext, fileFormat.toIceberg());
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        return writtenBytes;
+    }
+
+    @Override
+    public long getMemoryUsage()
+    {
+        return memoryUsage;
+    }
+
+    @Override
+    public long getValidationCpuNanos()
+    {
+        return validationCpuNanos;
+    }
+
+    @Override
+    public CompletableFuture<?> appendPage(Page page)
+    {
+        checkArgument(page.getChannelCount() == 1, "IcebergPositionDeletePageSink expected a Page with only one channel, but got " + page.getChannelCount());
+
+        Block[] blocks = new Block[2];
+        blocks[0] = new RunLengthEncodedBlock(nativeValueToBlock(VARCHAR, utf8Slice(dataFilePath)), page.getPositionCount());
+        blocks[1] = page.getBlock(0);
+        doAppend(new Page(blocks));
+
+        writtenData = true;
+        return NOT_BLOCKED;
+    }
+
+    @Override
+    public CompletableFuture<Collection<Slice>> finish()
+    {
+        Collection<Slice> commitTasks = new ArrayList<>();
+        if (writtenData) {
+            writer.commit();
+            CommitTaskData task = new CommitTaskData(
+                    outputPath,
+                    fileFormat,
+                    writer.getWrittenBytes(),
+                    new MetricsWrapper(writer.getMetrics()),
+                    PartitionSpecParser.toJson(partitionSpec),
+                    partition.map(PartitionData::toJson),
+                    FileContent.POSITION_DELETES,
+                    Optional.of(dataFilePath));
+            Long recordCount = task.getMetrics().recordCount();
+            if (recordCount != null && recordCount > 0) {
+                commitTasks.add(wrappedBuffer(jsonCodec.toJsonBytes(task)));
+            }
+            writtenBytes = writer.getWrittenBytes();
+            validationCpuNanos = writer.getValidationCpuNanos();
+        }
+        else {
+            // clean up the empty delete file
+            writer.rollback();
+        }
+        return completedFuture(commitTasks);
+    }
+
+    @Override
+    public void abort()
+    {
+        writer.rollback();
+    }
+
+    private void doAppend(Page page)
+    {
+        while (page.getPositionCount() > MAX_PAGE_POSITIONS) {
+            Page chunk = page.getRegion(0, MAX_PAGE_POSITIONS);
+            page = page.getRegion(MAX_PAGE_POSITIONS, page.getPositionCount() - MAX_PAGE_POSITIONS);
+            writePage(chunk);
+        }
+
+        writePage(page);
+    }
+
+    private void writePage(Page page)
+    {
+        long currentWritten = writer.getWrittenBytes();
+        long currentMemory = writer.getMemoryUsage();
+
+        writer.appendRows(page);
+
+        writtenBytes += (writer.getWrittenBytes() - currentWritten);
+        memoryUsage += (writer.getMemoryUsage() - currentMemory);
```

Comment:
why not just `memoryUsage = writer.getMemoryUsage()` (or remove the field)
same for `writtenBytes` ?

we don't have multiple writers, so we probably don't need these fields